Add prepare_fsd50k filenames object in metadata synthesizer

adrianSRoman opened the pull request:
-enable data synthesis with json data structure format

iranroman commented:
Thank you very much @adrianSRoman for creating this integration of your work to bring it to main. I have made some changes. The code seems to be fully functional. However, I have a few suggestions. Let me know what you think. @ChrisIck, @sivannavis  feel free to review and also let us know what you think.

iranroman commented on metadata_synthesizer.py:
```diff
@@ -64,15 +70,35 @@ def create_mixtures(self):
+                        audioclasses.extend([v]*len(vv))
+            audiofiles_ = []
+            for audiof in audiofiles:
+                if type(audiof) == list:
```
this is very awkward because we are keeping track of the duration of music files in addition to their path. Let's just keep the file path. For the time bring I think we should instead trim the music files (extra pre-processing step) and simply get their duration using `librosa.get_duration`. Alternatively we can add a duration during loading with `librosa.load`.

adrianSRoman replied:
Thanks @iranroman, after your changes things are much clearer. I added a small change to retrieve the tracks per classes in a single loop.

iranroman replied:
let's use these lines to standardize the durations to a fixed precision
```
for i in range(len(durations)): 
    durations[i] = np.floor(durations[i]*100)/100 
    onoffsets[i][1] = np.floor(onoffsets[i][1]*100)/100 
```

iranroman commented on audio_synthesizer.py:
```diff
@@ -119,7 +122,7 @@ def synthesize_mixtures(self):
                             eventsig, fs_db = soundfile.read(self._db_path + '/' + filename)   
                         elif self._db_name == 'fsd50k':
-                            eventsig, fs_db = soundfile.read(self._db_path + '/' + filename.split('/')[-1])
+                            eventsig, fs_db = librosa.load(filename, sr=self._fs, duration=onoffset[1]/10) # here we need librosa since we are loading .mp3 
```
I went ahead and added the duration argument to `librosa.load`

adrianSRoman replied:
It looks like librosa load takes a `float` that represents seconds. From the metadata I observe we set `[0, 10]`, which I assumed to be start at second `0` and end at second `10`. If we do `duration=onoffset[1]/10`, would this be trimming up to `1s` total per track?

iranroman replied:
the `duration` argument needs to be `(onoffset[1]-onoffset[0])-0.1`

iranroman replied:
eventsig, fs_db = librosa.load(path=self._db_path+'/'+filename.split('/')[-1], duration=(onoffset[1]-onoffset[0])-0.1)

iranroman commented on metadata_synthesizer.py:
```diff
@@ -64,15 +70,34 @@ def create_mixtures(self):
+            audiofiles_ = []
+            for audiof in audiofiles:
+                if type(audiof) == list:
+                    durations.append(librosa.get_duration(path=audiof[0]))
```
round down to the nearest tenth with `np.floor(dur*10)/10`

iranroman commented on metadata_synthesizer.py:
```diff
@@ -64,15 +70,34 @@ def create_mixtures(self):
+                    onoffsets.append([0,10])
+                    audiofiles_.append(audiof[0])
+                else:
+                    durations.append(librosa.get_duration(path=audiof))
```
round down to the nearest tenth with `np.floor(dur*10)/10`

iranroman commented on metadata_synthesizer.py:
```diff
@@ -64,15 +70,34 @@ def create_mixtures(self):
+            foldlist_nff['class'] = np.array(audioclasses)
+            foldlist_nff['audiofile'] = np.array(audiofiles_)
+            foldlist_nff['duration'] = np.array(durations)
+            foldlist_nff['onoffset'] = np.array(onoffsets)
```
here, actually:
```python
 for i in range(len(foldlist_nff['duration'])): 
    foldlist_nff['duration'][i] = np.floor(foldlist_nff['duration'][i]*100)/100 
    foldlist_nff['onoffset'][i][1] = np.floor(foldlist_nff['onoffset'][i][1]*100)/100 
```

iranroman commented:
Okay, I think this is pretty much ready to merge with `main`. We just need @sivannavis to validate the results with the generated data, and @adrianSRoman to update README.

adrianSRoman commented:
> Okay, I think this is pretty much ready to merge with `main`. We just need @sivannavis to validate the results with the generated data, and @adrianSRoman to update README.

Added some updates to the main README with a "one-time setup" section. Let me know what you all think about this.

I also added a detailed description to this PR. @sivannavis let me know if things are clear for you to start working on this. You will be the second person using this, so any feedback on code and documentation would be greatly appreciated!

ChrisIck commented on README.md:
````diff
@@ -1,34 +1,61 @@
 
 ```
-pip install mirdata
+PARAM_CONFIG = {
````
Ideally this will be pulled from `.yaml` down the line, but this is good for now.

ChrisIck commented:
All of this LGTM. The only comment I have is that at this point, with the FSD50k download/prep and music trimming, we may want to make a singular automated script to make the setup process easier (the `README.md` file is getting long). Let's discuss further.
